src/prettytable/prettytable.py

```diff
@@ -1030,13 +1060,13 @@ def vertical_char(self):
         return self._vertical_char
 
     @vertical_char.setter
-    def vertical_char(self, val) -> None:
+    def vertical_char(self, val: str) -> None:
         val = str(val)
         self._validate_option("vertical_char", val)
```

Comment:
A lot of these attribute setters could work with a wider range of values. To be useful for type checkers, I think it makes sense to use the closest matching type here though.

E.g. even if an `int` would work with `str(1)`, it makes sense if type checkers are able to warn if it isn't a string which is passed to the attribute setter.